### PR TITLE
Issue #1142 - move code validation logic to DefaultTermServiceProvider

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: fhir-term/src/main/java/com/ibm/fhir/term/spi/FHIRTermServiceProvider.java [hunk 1]
```diff
@@ -67,6 +68,22 @@
      */
     boolean validateCode(Coding coding);
 
+    /**
+     * Indicates whether the given code is a member of the specified value set
+     *
+     * @param valueSet
+     *     the value set
```
Comment (lmsurpre) [inline fhir-term/src/main/java/com/ibm/fhir/term/spi/FHIRTermServiceProvider.java:75]: an external contract about whether the valueset must be pre-expanded?

File: fhir-term/src/main/java/com/ibm/fhir/term/spi/FHIRTermServiceProvider.java [hunk 2]
```diff
@@ -78,4 +95,16 @@
      *     true if the given coding is a member of the specified value set, false otherwise
      */
     boolean validateCode(ValueSet valueSet, Coding coding);
+
+    /**
+     * Indicates whether the given codeable concept contains a coding that is a member of the specified value set
+     *
+     * @param valueSet
+     *     the value set
```
Comment (lmsurpre) [inline fhir-term/src/main/java/com/ibm/fhir/term/spi/FHIRTermServiceProvider.java:103]: any external contract about whether the valueset must be pre-expanded?
assuming not, is it worth mentioning that the service should expand it if needed?

Comment (JohnTimm) [reply fhir-term/src/main/java/com/ibm/fhir/term/spi/FHIRTermServiceProvider.java:109]: Added an `@apiNote` comment to the Javadoc

File: fhir-term/src/main/java/com/ibm/fhir/term/spi/FHIRTermServiceProvider.java (hunk 1 above)
Comment (JohnTimm) [reply fhir-term/src/main/java/com/ibm/fhir/term/spi/FHIRTermServiceProvider.java:77]: Added an `@apiNote` comment to the Javadoc

**Pull request merged**